Add STG to community pipelines

kinam0252 opened the pull request:
# What does this PR do?

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

### Spatiotemporal Skip Guidance

**Junha Hyung\*, Kinam Kim\*, Susung Hong, Min-Jung Kim, Jaegul Choo**

**KAIST AI, University of Washington**

[*Spatiotemporal Skip Guidance (STG) for Enhanced Video Diffusion Sampling*](https://arxiv.org/abs/2411.18664) (CVPR 2025) is a simple training-free sampling guidance method for enhancing transformer-based video diffusion models. STG employs an implicit weak model via self-perturbation, avoiding the need for external models or additional training. By selectively skipping spatiotemporal layers, STG produces an aligned, degraded version of the original model to boost sample quality without compromising diversity or dynamic degree.

Following is the example video of STG applied to Mochi.


https://github.com/user-attachments/assets/f170c3db-8e46-482c-8199-30d3ebcb6c0f


More examples and information can be found on the [GitHub repository](https://github.com/junhahyung/STGuidance) and the [Project website](https://junhahyung.github.io/STGuidance/).


## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sayakpaul 

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @.

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Core library:

- Schedulers: @yiyixuxu
- Pipelines and pipeline callbacks: @yiyixuxu and @asomoza
- Training examples: @sayakpaul
- Docs: @stevhliu and @sayakpaul
- JAX and MPS: @pcuenca
- Audio: @sanchit-gandhi
- General functionalities: @sayakpaul @yiyixuxu @DN6

Integrations:

- deepspeed: HF Trainer/Accelerate: @SunMarc
- PEFT: @sayakpaul @BenjaminBossan

HF projects:

- accelerate: [different repo](https://github.com/huggingface/accelerate)
- datasets: [different repo](https://github.com/huggingface/datasets)
- transformers: [different repo](https://github.com/huggingface/transformers)
- safetensors: [different repo](https://github.com/huggingface/safetensors)

-->

a-r-r-o-w reviewed (approved):
Thank you very much for the PR! This is really amazing work and the paper was very intuitive to understand 🤗 

I believe the example docstrings are out of sync with the pipelines. Could those be updated to reflect usage? I haven't exhaustively tested all pipelines, but looks good to me from the code.

This is very high up on my priority list too to integrate in core diffusers, and I'm working on adding these methods and make available to all models for #9672!

kinam0252 commented:
Thank you for your kind words!😊I really appreciate your feedback. 

I've updated the example docstrings to include sample usage. Let me know if there's anything else you'd like to adjust!
I’m also looking forward to seeing our method integrated into core diffusers! 🚀

a-r-r-o-w commented:
@kinam0252 Looks good to me. I believe the imports and pipeline names are incorrect, which could be updated.

The failing tests seem to be because of a typo: 
![image](https://github.com/user-attachments/assets/c178af9e-da60-4214-b632-706c9aae9e1c)

Could you update this as well? Will be good to merge after that 🤗

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/diffusers/pr_10960). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

kinam0252 commented:
@a-r-r-o-w Sorry for the previous mistakes! I’ve updated the imports, fixed the pipelines, and corrected the typos. 
Hopefully, this resolves all the issues. 🚀 Please let me know if anything else needs to be fixed.

a-r-r-o-w commented:
Thanks @kinam0252! Looks good to merge

@bot /style

a-r-r-o-w commented:
@kinam0252 Is rescaling supported in HunyuanVideoSTGPipeline? The style bot seems to be failing for this as it is undefined: https://github.com/huggingface/diffusers/actions/runs/13724296073/job/38386911164

![image](https://github.com/user-attachments/assets/7ddcfc87-2d36-45e9-a79b-ff4c070bda29)

kinam0252 commented:
@a-r-r-o-w Oh the style bot is right - rescaling is not supported in HunyuanVideoSTGPipeline. I just fixed it!

a-r-r-o-w commented:
@bot /style

kinam0252 commented:
Thank you so much! 😊
